metricflow/model/transformations/dbt_to_metricflow.py

```diff
@@ -339,7 +342,7 @@ def collect_time_dimension_stats(cls, dbt_metrics: List[DbtMetric]) -> Dict[str,
         # dimenson for the data source that is built for the `DbtMetric.model`
         time_stats_for_metric_models: Dict[str, Dict[str, int]] = {}
         for dbt_metric in dbt_metrics:
-            if dbt_metric.calculation_method != "derived":
+            if dbt_metric.calculation_method != "derived" and dbt_metric.model is not None:
```

Comment:
I believe all dbt metrics that aren't `derived` must have a `model`. However, I'm not sure if dbt enforces this when parsing the dbt project / building the manifest. Thus although we likely don't need the extra check, it doesn't hurt.